src/models/schemaRegistry.ts

```diff
@@ -1,53 +1,58 @@
 import { Data, type Require as Enforced } from "dataclass";
-import * as vscode from "vscode";
+import { MarkdownString, ThemeIcon, TreeItem, TreeItemCollapsibleState } from "vscode";
+import { ConnectionType } from "../clients/sidecar";
 import { CCLOUD_CONNECTION_ID, IconNames, LOCAL_CONNECTION_ID } from "../constants";
 import { CustomMarkdownString } from "./main";
+import { ConnectionId, IResourceBase, isCCloud } from "./resource";
 
-export abstract class SchemaRegistry extends Data {
-  abstract readonly connectionId: string;
-  abstract readonly isLocal: boolean;
-  abstract readonly isCCloud: boolean;
+export abstract class SchemaRegistry extends Data implements IResourceBase {
+  abstract connectionId: ConnectionId;
+  abstract connectionType: ConnectionType;
+  iconName: IconNames = IconNames.SCHEMA_REGISTRY;
 
   id!: Enforced<string>;
   uri!: Enforced<string>;
-}
-
-export class LocalSchemaRegistry extends SchemaRegistry {
-  readonly connectionId = LOCAL_CONNECTION_ID;
-  readonly isLocal: boolean = true;
-  readonly isCCloud: boolean = false;
+  // added separately from sidecar responses
+  environmentId!: Enforced<string>;
 }
 
 export class CCloudSchemaRegistry extends SchemaRegistry {
-  readonly connectionId = CCLOUD_CONNECTION_ID;
-  readonly isLocal: boolean = false;
-  readonly isCCloud: boolean = true;
+  readonly connectionId: ConnectionId = CCLOUD_CONNECTION_ID as ConnectionId;
+  readonly connectionType: ConnectionType = ConnectionType.Ccloud as ConnectionType;
 
   provider!: Enforced<string>;
   region!: Enforced<string>;
-  // added separately from sidecar responses
-  environmentId!: Enforced<string>;
 
   get ccloudUrl(): string {
     return `https://confluent.cloud/environments/${this.environmentId}/schema-registry/schemas`;
   }
 }
 
-// Tree item representing a Schema Registry in the Resources view
-export class SchemaRegistryTreeItem extends vscode.TreeItem {
+export class DirectSchemaRegistry extends SchemaRegistry {
+  connectionId!: ConnectionId;
```

Comment:
Why no Enforced<> ? And prob better to get the readonly attrs defined above the actual settable ones?